http/file_server_test.ts

```diff
@@ -540,31 +545,22 @@ Deno.test(
   async function () {
     await startTlsFileServer();
     try {
-      // Valid request after invalid
-      const conn = await Deno.connectTls({
-        hostname: "localhost",
-        port: 4577,
-        certFile: join(testdataDir, "tls/RootCA.pem"),
-      });
-
-      await writeAll(
-        conn,
-        new TextEncoder().encode("GET / HTTP/1.0\r\n\r\n"),
+      const caCert = await Deno.readTextFile(
+        join(testdataDir, "tls/RootCA.pem"),
       );
-      const res = new Uint8Array(128 * 1024);
-      const nread = await conn.read(res);
-      assert(nread !== null);
-      conn.close();
-      const page = new TextDecoder().decode(res.subarray(0, nread));
-      assert(page.includes("<title>Deno File Server</title>"));
+      const client = Deno.createHttpClient({ caCerts: [caCert] });
+      const res = await fetch("https://localhost:4577/", { client });
+      client.close();
+
+      assertStringIncludes(await res.text(), "<title>Deno File Server</title>");
     } finally {
       await killFileServer();
     }
   },
 );
 
 Deno.test("partial TLS arguments fail", async function () {
```

Comment:
I don't think we need to change this test case because according to this run log https://github.com/denoland/deno_std/actions/runs/6625309968/job/17996042005?pr=3715 this test case was affected by the one above (i.e. the flakiness wasn't caused by this test case itself)

Reply:
I understand. However, passing by, I realised the test could rely on more reliable APIs. Hence, the change.